beacon-chain/db/kv/kv.go

```diff
@@ -229,21 +231,33 @@ func (s *Store) DatabasePath() string {
 	return s.databasePath
 }
 
-func (s *Store) checkNeedsResync() error {
-	return s.db.View(func(tx *bolt.Tx) error {
-		bkt := tx.Bucket(migrationsBucket)
-		hasDisabledFeature := !features.Get().EnableOnlyBlindedBeaconBlocks
-		if hasDisabledFeature && bkt.Get(migrationBlindedBeaconBlocksKey) != nil {
-			return fmt.Errorf(
-				"you have disabled the flag %s, and your node must resync to ensure your "+
-					"database is compatible. If you do not want to resync, please re-enable the %s flag",
-				features.EnableOnlyBlindedBeaconBlocks.Name,
-				features.EnableOnlyBlindedBeaconBlocks.Name,
-			)
+func (s *Store) setupBlockStorageType(isFreshDatabase bool) error {
```

Comment:
This function doesn't handle this particular case: 
- User has been running with `--enable-only-blinded-beacon-blocks` since bellatrix.
- The above flag has now been deprecated, so when a user running with an old database runs the new prysm version, they will start saving full blocks into the db (rather than blinded or what they were doing before).

Basically we need  to find a way to insert `saveBlindedBeaconBlocksKey` into the chain metadata bucket for users who were already running with blinded blocks.

Reply:
Nice catch! Added @nisdas